### PR TITLE
remove lotus filecoin storage and simplify publishers

### PR DESCRIPTION
**Pull request by frrist**
- closes https://github.com/bacalhau-project/bacalhau/issues/2538 and closes https://github.com/bacalhau-project/bacalhau/issues/1705
- Wrote https://github.com/bacalhau-project/bacalhau/issues/2555 while implementing. This is a fix for a follow on.

### CODE REVIEW

File: pkg/model/publisher.go [hunk 1]
```diff
@@ -11,7 +11,6 @@ const (
 	publisherUnknown Publisher = iota // must be first
 	PublisherNoop
 	PublisherIpfs
-	PublisherFilecoin
```
Comment (simonwo) [inline pkg/model/publisher.go:14]: Removing this means that people who have in the past submitted Estuary or S3 jobs now have data that is interpreted incorrectly, e.g. previously Estuary was 4 but now S3 has that value, so places where the numbers are used (existing data) will be wrong. Can we be sure that noone is using these numbers? If not, can we just use `PublisherEstuary = 4` and make sure that 3 isn't used?

File: pkg/model/schemas/bacalhau.ipldsch [hunk 2]
```diff
@@ -36,7 +36,6 @@ type Publisher enum {
     | Noop ("1")
     | IPFS ("2")
     | Estuary ("3")
-    | FilecoinLotus ("4")
 } representation int
```
Comment (simonwo) [inline pkg/model/schemas/bacalhau.ipldsch:39]: Lol this was wrong? Can we correct the value for Estuary here?

File: pkg/model/publisher.go (hunk 1 above)
Comment (frrist) [reply pkg/model/publisher.go:14]: In a very similar vein to https://github.com/bacalhau-project/bacalhau/pull/2552#discussion_r1224572846: These will go away when https://github.com/bacalhau-project/bacalhau/issues/2543 lands. 

The pattern of using enums/constants for these values comes at the cost of them being hard to change later, the work to complete #2543 will replace these with CID's backed by a schema similar to how https://github.com/bacalhau-project/bacalhau/issues/2542 will replace engine enums with CIDs.

Comment (wdbaruni) [reply pkg/model/publisher.go:14]: This shouldn't be a problem as we always marshal and unmarshal publisher types to their string value before sending them around

**Review by wdbaruni — APPROVED** [review]

**Pull request merged**